packages/marshal/src/helpers/safe-promise.js

```diff
@@ -23,25 +23,52 @@ const checkPromiseOwnKeys = (pr, check) => {
     key => typeof key !== 'symbol' || !hasOwnPropertyOf(Promise.prototype, key),
   );
 
+  const checkSafeEnoughKey = key => {
```

Comment:
Nit: maybe we should be explicit in the function name that we're checking async hooks values?

Reply:
Unrelated, I don't know how I feel about allocating a closure for something which really should seldomly be encountered. I kinda would prefer if this was either inline in the `every`, or factored out of the checkPromiseOwnKeys function, but I'm probably prematurely optimizing.

Reply:
`checkSafeAsyncHooksKey`.

Kept it factored out but refactored the context so you get your premature optimization anyway. I think a bit better even ;)

Reply:
Done